tests/integration/base.py

```diff
@@ -373,6 +374,38 @@ def generate_and_settle(self, wait_interfaces=None, state_dir=None):
 
         if 'Run \'systemctl daemon-reload\' to reload units.' in out:
             print('\nWARNING: systemd units changed without reload:', out)
+
+        self.settle(wait_interfaces, state_dir)
+
+    def try_and_settle(self, try_func, wait_interfaces_try=None, wait_interfaces_after=None):
+        '''Run `netplan try`, wait for the configuration to apply, run `try_func` and allow try to time out'''
+
+        # Because `netplan try` uses select(2) to implement the timeout, stdin
+        # needs to be a stream that is never ready to read.
+        fifo_path = f"/tmp/netplan-try-test.{random.randint(10000, 99999)}"
```

Comment:
though (non-blocking): This could introduce a flaky failure if the path already exists (chances are rather low). I wonder if we could use a higher level function, such as https://docs.python.org/3/library/tempfile.html#tempfile.mkstemp, removing the created tempfile right afterward and re-using the path for our FIFO.